packages/bp/src/admin/workspace/collaborators/collaborators-router.ts

```diff
@@ -156,11 +174,24 @@ class CollaboratorsRouter extends CustomAdminRouter {
 
     router.get(
       '/reset/:strategy/:email',
-      assertSuperAdmin,
-      this.needPermissions('write', this.resource),
       this.asyncMiddleware(async (req, res) => {
         const { email, strategy } = req.params
 
+        const user = await this.workspaceService.findUser(
+          req.params.email,
+          req.params.strategy,
+          req.workspace || 'default'
+        )
+        if (!req.tokenUser?.isSuperAdmin) {
+          const mwHandler = this.needPermissions('write', `admin.collaborators.${user?.role}`)
+          try {
+            await Promise.fromCallback(cb => mwHandler(req, res, cb))
+          } catch (err) {
+            res.sendStatus(403)
+            return
+          }
+        }
```

Comment:
You could write a middleware instead of repeating this code, ex:
```js
async assertUserAccess(req: Request, res: Response, next: Function) {
    const { tokenUser, workspace } = <RequestWithUser>req
    if (!tokenUser || !workspace) {
      return next(new InternalServerError('Missing informations'))
    }

    if (tokenUser.isSuperAdmin) {
      return next()
    }

    const targetUser = await this.workspaceService.findUser(req.params.email, req.params.strategy, workspace)
    if (targetUser) {
      return next(this.needPermissions('write', `admin.collaborators.${targetUser.role}`))
    }

    next(new InternalServerError('No access to user'))
  }
```

Maybe `admin.collaborators.manage.${ROLE}` would be a better naming? 

While were at it, maybe we could add more fine-grained properties (eg: collaborators.addUserToWorkspace, deleteFromWorkspace, updateRole, etc...) since now we only seems to have a wide access to all collaborators methods